modules/system/rolls/cast-test.js

```diff
@@ -9,6 +9,10 @@ export default class CastTest extends TestWFRP {
 
     this.preData.itemData = data.itemData || this.item.toObject() // Store item data to avoid rerolls being affected by changed channeled SL
     this.preData.skillSelected = data.skillSelected;
+    this.preData.overchannelling = data.overchannelling;
+    this.preData.quickcasting = data.quickcasting;
+    this.preData.unofficialGrimoire = data.unofficialGrimoire;
+    this.preData.ingredientMode = data.ingredientMode;
```

Comment:
Isn't this all unofficial grimoire? so an unofficialGrimoire object that holds all these flags should work?

Reply:
done, changed into object